iothub/device/src/ClientFactory.cs

```diff
@@ -100,7 +100,7 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
                     throw new ArgumentException("No certificate was found. To use certificate authentication certificate must be present.");
                 }
 
-                InternalClient dc = CreateFromConnectionString(connectionStringBuilder.ToString(), PopulateCertificateInTransportSettings(connectionStringBuilder, transportType), options);
+                using InternalClient dc = CreateFromConnectionString(connectionStringBuilder.ToString(), PopulateCertificateInTransportSettings(connectionStringBuilder, transportType), options);
```

Comment:
this client shouldn't be disposed here, it is returned to the caller, and they should dispose it

Reply:
Makes sense. So I guess we need to ignore the warning in this case.